src/Microsoft.AspNet.TestHost/ClientHandler.cs

```diff
@@ -121,6 +124,12 @@ internal RequestState(HttpRequestMessage request, PathString pathBase)
                 }
 
                 HttpContext = new DefaultHttpContext();
+
+                if (accessor != null)
```

Comment:
I'm assuming this is because we need to decide whether to use the default factory or keep this hack for the custom factory?